irohad/ametsuchi/impl/postgres_query_executor.cpp

```diff
@@ -312,19 +313,34 @@ namespace iroha {
         log_->error("query signatories did not pass validation");
         return false;
       }
+      if (not visitor_.hasAccountRolePermission(Role::kGetBlocks,
+                                                query.creatorAccountId())) {
+        log_->error("query creator does not have enough permissions");
+        return false;
+      }
+
+      return true;
+    }
+
+    bool PostgresQueryExecutorVisitor::hasAccountRolePermission(
+        shared_model::interface::permissions::Role permission,
+        const std::string &account_id) const {
       using T = boost::tuple<int>;
       boost::format cmd(R"(%s)");
       try {
         soci::rowset<T> st =
-            (sql_->prepare
-                 << (cmd % checkAccountRolePermission(Role::kGetBlocks)).str(),
-             soci::use(query.creatorAccountId(), "role_account_id"));
+            (sql_.prepare
+                 << (cmd % getAccountRolePermissionCheckSql(permission)).str(),
+             soci::use(account_id, "role_account_id"));
 
-        return st.begin()->get<0>();
```

Comment:
`return st.begin()->get<0>();` seems to work well here, so the change does not seem necessary.